python/cuml/dask/linear_model/linear_regression.py

```diff
@@ -14,547 +13,315 @@
         self._model_fit = False
         self._consec_call = 0
 
-    def _build_params_map(self):
-        return {"fit_intercept": self.fit_intercept,
-                "normalize": self.normalize}
-
-    def fit(self, X, y):
+    @staticmethod
+    def _func_create_model(sessionId, **kwargs):
+        try:
+            from cuml.linear_model.linear_regression_mg \
+               import LinearRegressionMG as cumlLinearRegression
+        except ImportError:
+            raise Exception("cuML has not been built with multiGPU support "
+                            "enabled. Build with the --multigpu flag to"
+                            " enable multiGPU support.")
+
+        handle = worker_state(sessionId)["handle"]
+        return cumlLinearRegression(handle=handle, **kwargs)
+
+    @staticmethod
+    def _func_fit_colocated(f, data, M, N, partsToRanks, rank):
+        return f.fit_colocated(data, M, N, partsToRanks, rank)
+
+    @staticmethod
+    def _func_fit(f, X, y, M, N, partsToRanks, rank):
+        return f.fit(X, y, M, N, partsToRanks, rank)
+
+    @staticmethod
+    def _func_predict(f, df, M, N, partsToRanks, rank):
+        return f.predict(df, M, N, partsToRanks, rank)
+
+    @staticmethod
+    def _func_get_first(f):
+        return f[0]
+
+    @staticmethod
+    def _func_get_idx(f, idx):
+        return f[idx]
+
+    @staticmethod
+    def _func_xform(model, df):
+        return model.transform(df)
+
+    @staticmethod
+    def _func_get_size_cl(df):
+        return df[0].shape[0]
+
+    @staticmethod
+    def _func_get_size(df):
+        return df.shape[0]
+
+    def _fit(self, X, y):
+        X_futures = self.client.sync(extract_ddf_partitions, X)
+        y_futures = self.client.sync(extract_ddf_partitions, y)
+
+        X_partition_workers = [w for w, xc in X_futures]
+        y_partition_workers = [w for w, xc in y_futures]
+
+        if set(X_partition_workers) != set(y_partition_workers):
+            raise ValueError("""
+              X  and y are not partitioned on the same workers expected \n
+              Linear Regression""")
+
+        self.rnks = dict()
+        rnk_counter = 0
+        worker_to_parts = OrderedDict()
+        for w, p in X_futures:
+            if w not in worker_to_parts:
+                worker_to_parts[w] = []
+            if w not in self.rnks.keys():
+                self.rnks[w] = rnk_counter
+                rnk_counter = rnk_counter + 1
+            worker_to_parts[w].append(p)
+
+        worker_to_parts_y = OrderedDict()
+        for w, p in y_futures:
+            if w not in worker_to_parts_y:
+                worker_to_parts_y[w] = []
+            worker_to_parts_y[w].append(p)
+
+        workers = list(map(lambda x: x[0], X_futures))
+
+        comms = CommsContext(comms_p2p=False)
+        comms.init(workers=workers)
+
+        worker_info = comms.worker_info(comms.worker_addresses)
+
+        key = uuid1()
+        partsToRanks = [(worker_info[wf[0]]["r"], self.client.submit(
```

Comment:
As noted in Ridge, I think this should be called partsToSizes?

Reply:
Done.